packages/pages/src/upgrade/pagesUpdater.test.ts

```diff
@@ -1,57 +1,56 @@
 import { describe, it, expect } from "vitest";
 import { removeFetchImport } from "./pagesUpdater.js";
-import { Project } from "ts-morph";
 import path from "path";
 import fs from "fs";
-import typescript from "typescript";
 
-describe("test pages updater steps", () => {
-  it("removes fetch imports", () => {
-    const expectedBefore =
-      'import { fetch, tacos } from "@yext/pages/util";\n' +
-      'import fetch from "@fetch/fetcher";\n';
-    const expectedAfter =
-      'import { tacos } from "@yext/pages/util";\n' +
-      'import fetch from "@fetch/fetcher";\n';
-
-    // generate test file
-    const testFilePath = "src/upgrade/test/src/templates/location.tsx";
-    const source = path.resolve("src/upgrade/test/src");
-    const testProject = new Project({
-      compilerOptions: {
-        jsx: typescript.JsxEmit.ReactJSX,
-        sourceRoot: source,
-      },
-    });
-    testProject.addSourceFilesAtPaths([
-      `${source}/**/*.ts`,
-      `${source}/**/*.tsx`,
-      `${source}/**/*.js`,
-      `${source}/**/*.jsx`,
-    ]);
-    let sourceFile = testProject.addSourceFileAtPathIfExists(testFilePath);
-    if (!sourceFile) {
-      sourceFile = testProject.createSourceFile(testFilePath);
-    }
-    sourceFile.addImportDeclaration({
-      namedImports: ["fetch", "tacos"],
-      moduleSpecifier: "@yext/pages/util",
-    });
-    sourceFile.addImportDeclaration({
-      defaultImport: "fetch",
-      moduleSpecifier: "@fetch/fetcher",
-    });
-    testProject.saveSync();
-    const beforeContent = fs.readFileSync(path.resolve(testFilePath), "utf-8");
-
-    // modify test file
-    removeFetchImport(source);
+function testFetchRemoval(
+  fileName: string,
+  beforeContent: string,
+  expected: string
+): boolean {
+  const testFolder = path.resolve("src/upgrade/test");
+  const source = path.resolve(testFolder, "src");
+  const templates = path.resolve(source, "templates");
+  fs.mkdirSync(templates, {
+    recursive: true,
+  });
+  const filePath = path.resolve(templates, fileName);
+  fs.writeFileSync(filePath, beforeContent);
+  removeFetchImport(source);
```

Comment:
Can you not pass ts-morph a string directly instead of having it read an actual file? It's a little wonky to be creating and deleting files for the tests.

Reply:
I wasn't able to find a way to enforce line length. Multiple imports all ended up on one line.

The previous method with ts-morph was also creating and deleting files.

Reply:
The function we are testing is also just given a source path, then it creates a ts-morph project to read it. We would have to change the function in order to pass it a string without creating a file.

Reply:
That's what I was thinking we'd do, but it's not all that important. We can leave how it is.